src/sentry/api/serializers/models/incident.py

```diff
@@ -29,8 +37,33 @@ def get_attrs(self, item_list, user, **kwargs):
             results[incident] = {"projects": incident_projects.get(incident.id, [])}
             results[incident]["alert_rule"] = alert_rules.get(str(incident.alert_rule.id))
 
+            if "activities" in self.expand:
+                results[incident]["activities"] = serialize(
+                    list(
+                        IncidentActivity.objects.filter(incident=incident).select_related(
+                            "user", "incident"
+                        )
+                    )
+                )
```

Comment:
This query needs to be done in bulk - outside of this loop. So you'd do `IncidentActivity.objects.filter(incident__in=item_list)` to fetch all the activities and also serialize them out there. A good example of how to do this well is `IncidentProject` a few lines up.